admin/app/controllers/spree/admin/payments_controller.rb

```diff
@@ -10,10 +10,12 @@ def new
         payment_method = params[:payment_method_id] ? @payment_methods.find { |pm| pm.id.to_s == params[:payment_method_id].to_s } : @payment_methods.first
 
         @payment = @order.payments.build(
-          amount: @order.total_minus_store_credits - @order.payment_total,
+          amount: @order.total - @order.payment_total,
           payment_method: payment_method
         )
 
+        @payment_max_amount = payment_method.store_credit? ? [@user_store_credits.first.amount_remaining, @payment.amount].min : @payment.amount
```

Comment:
@mad-eel let's add `Payment#max_amount` method, also we need validation in model